include/openssl/ssl.h

```diff
@@ -1949,6 +1950,18 @@ void SSL_set_not_resumable_session_callback(SSL *ssl,
                                             int (*cb) (SSL *ssl,
                                                        int
                                                        is_forward_secure));
+
+void SSL_CTX_set_choose_cipher_callback(SSL_CTX *ctx,
+                                        const SSL_CIPHER* (*cb) (SSL *ssl,
+                                                                 STACK_OF(SSL_CIPHER) *client));
+
+void SSL_set_choose_cipher_callback(SSL *ssl,
+                                    const SSL_CIPHER* (*cb) (SSL *ssl,
+                                                             STACK_OF(SSL_CIPHER) *client));
+
+
+const SSL_CIPHER *SSL_choose_cipher_default(SSL *s, STACK_OF(SSL_CIPHER) *clnt);
```

Comment:
The point is to allow a consumer to have some simple override logic, then fall through to the default if it's not applicable.